src/frontend/src/catalog/database_catalog.rs

```diff
@@ -99,10 +99,23 @@ impl DatabaseCatalog {
             .find(|schema| schema.get_table_by_id(table_id).is_some())
     }
 
-    pub fn update_self(&mut self, prost: &PbDatabase) {
-        self.id = prost.id;
-        self.name = prost.name.clone();
-        self.owner = prost.owner;
+    pub fn update_schema(&mut self, prost: &PbSchema) {
+        let id = prost.id;
+        let name = prost.name.clone();
+
+        let schema = self.get_schema_by_id(&id).unwrap();
+        let old_schema_name = schema.name();
+        let mut schema = if old_schema_name != name {
+            self.schema_by_name.remove(&old_schema_name).unwrap()
+        } else {
+            schema.clone()
```

Comment:
Nits: can we use `get_mut` to change the fields directly rather than clone it? After that we can also get the old schema name in `schema_name_by_id`.